cmake/templates/BoostConfig.cmake.in

```diff
@@ -51,10 +50,16 @@ if(NOT TARGET "Boost::boost")
       INTERFACE_INCLUDE_DIRECTORIES "${_root_path}/include"
   )
   if(MSVC)
-    set_target_properties(
+    #stop MSVC from trying to auto-link libs
+	set(compile_definitions "BOOST_ALL_NO_LIB=1")
+    if(@BOOST_ALL_DYN_LINK@) #required for MSVC when using shared libs
+	  set(compile_definitions "${compile_definitions} BOOST_ALL_DYN_LINK")
```

Comment:
You are correct it is **not** what we are expecting, I was thinking cl.exe was interpreting (through a "bug") /D "BOOST_ALL_NO_LIB=1 BOOST_ALL_DYN_LINK" as /D "BOOST_ALL_NO_LIB=1" /D"BOOST_ALL_DYN_LINK" and that was the reason the compile was working. But it seems it is just ignored and only sees /D"BOOST_ALL_NO_LIB=1". The compile is working because BOOST_ALL_DYN_LINK only turns on __declspec(dllimport) for functions and MSVC does not need dllimport declared to compile, although for performance/code size it should be.:

>https://msdn.microsoft.com/en-us/library/aa271769(v=vs.60).aspx
>You do not need to use __declspec(dllimport) for your code to compile correctly, but doing so allows the compiler to generate better code. The compiler is able to generate better code because it knows for sure whether a function exists in a DLL or not, so the compiler can produce code that skips a level of indirection that would normally be present in a function call that crossed a DLL boundary. However, you must use __declspec(dllimport) in order to import variables used in a DLL.

So it will work without BOOST_ALL_DYN_LINK but I think we should really figure out a way to include it, any ideas?